agent/engine/docker_task_engine.go

```diff
@@ -447,8 +448,19 @@ func (engine *DockerTaskEngine) AddTask(task *api.Task) error {
 
 	existingTask, exists := engine.state.TaskByArn(task.Arn)
 	if !exists {
+		// This will update the container desired status
+		task.UpdateDesiredStatus()
+
 		engine.state.AddTask(task)
-		engine.startTask(task)
+		if dependencygraph.ValidDependencies(task) {
+			engine.startTask(task)
+		} else {
+			seelog.Errorf("Task can not move forward due to the dependencies of containers cannot be resolved, task:%s", task.String())
```

Comment:
We should either use `can not` or `cannot`. Let's not use both. I'd vote for `cannot`. Also, can you leave a space after `task: `?